bytecode/flags.py

```diff
@@ -0,0 +1,297 @@
+                        else:
+                            flags[k] = False
+                    elif k == 'nofree':
+                        flags[k] = not (instr_names & {'LOAD_CLOSURE',
+                                                       'LOAD_DEREF',
+                                                       'STORE_DEREF',
+                                                       'DELETE_DEREF',
+                                                       'LOAD_CLASSDEREF'})
+                    else:
+                        if not any((self.async_generator,
+                                    self.iterable_coroutine)):
+                            flags[k] = bool(instr_names & {'GET_AWAITABLE',
+                                                           'GET_AITER',
+                                                           'GET_ANEXT',
+                                                           'BEFORE_ASYNC_WITH',
+                                                           'SETUP_ASYNC_WITH'})
+                        else:
+                            flags[k] = False
+                else:
+                    flags[k] = v
+
+        if [flags[k] for k in ('coroutine', 'iterable_coroutine', 'generator',
+                               'async_generator')].count(True) > 1:
+            raise ValueError("Code should not have more than one of the "
+                             "following flag set : generator, coroutine, "
+                             "iterable coroutine and async generator")
+
+        return sum(flags[k] and getattr(CoFlags, 'CO_' + k.upper())
+                   for k in flags)
+
+    def get_default(self, flag_name):
+        return self._defaults[flag_name]
+
+    @property
+    def optimized(self):
+        """Denotes that the code block uses fast locals.
+
+        """
+        return self._forced.get('optimized')
+
+    @optimized.setter
+    def optimized(self, value):
+        if value is None:
+            del self._forced['optimized']
+        elif not isinstance(value, bool):
+            raise ValueError('Flag value should be a boolean or None')
+        else:
+            self._forced['optimized'] = value
+
+    @property
+    def newlocals(self):
+        """Denotes that a new dictionary should be created for the code block.
+
+        """
+        return self._forced.get('newlocals', self._defaults['newlocals'])
+
+    @newlocals.setter
+    def newlocals(self, value):
+        if not isinstance(value, bool):
+            raise ValueError('Flag value should be a boolean')
+        self._forced['newlocals'] = value
+
+    @property
+    def varargs(self):
+        """The compiled code block has a varargs argument.
+
+        """
+        return self._forced.get('varargs', self._defaults['varargs'])
+
+    @varargs.setter
+    def varargs(self, value):
+        if not isinstance(value, bool):
+            raise ValueError('Flag value should be a boolean')
+        self._forced['varargs'] = value
```

Comment:
Maybe you can factorize these methods with __getattr__() and __setattr__() methods, the code looks to be duplicated for each flag.